### DISCUSSION TITLE
reset api with react-select

### DISCUSSION THREAD

**Discussion by andrestrei** [react-hook-form/react-hook-form#8423] [Q&A] [answered]
Using RHF with react-select, yup and typescript. I'm having trouble using `reset` API with `react-select` and defaultValues.  Basically I want my default value to be that nothing is selected (want to see the placeholder text from react-select). If I use `defaultValue={undefined}` or no defaultvalue at all, after I reset the underlying select value is cleared in RHF but the UI still shows the previously selected value and not the placeholder text from react-select. 
https://codesandbox.io/s/icy-river-ddx2v3

**Comment by bluebill1049** [comment]
not support `undefined` for defaultValue, this apply to controlled component as well.

**Comment by leapful** [comment, marked as answer]
@andrestrei @bluebill1049 : According to react-select documentation, clearing value should be done by calling method "clearValue" of the Select component instance.

```
const ref = useRef();

<Select ref={(instance) => {
   ref.current = instance;
}} />

// then just call method "clearValue" when resetting the form.

const onReset = () => {
  ref.current.clearValue();
  // reset() of react-hook-form
  reset();
}
```

Working demo: https://codesandbox.io/s/hidden-morning-qfqxgv?file=/src/App.tsx

**Reply by bluebill1049** [reply]
Thanks @leapful 🙏

**Reply by andrestrei** [reply]
that did the trick, TY
